### ISSUE TITLE
Add Pod TimeOut configurator to `odo utils config` / fix current e2e tests

### ISSUE THREAD

**Issue by cdrage** [openshift/odo#943]
So I figured out what's happening with tests currently failing. Essentially:

1. We run the tests
2. The tests take wayyyy too long since we're maxing out TravisCI
3. Depending on TravisCI's workload / bandwidth / internet it may or my not fail.

To fix:
As per this error: https://travis-ci.org/redhat-developer/odo/jobs/450912699#L815

The fix is actually to *increase* the PodTimeout:

https://github.com/redhat-developer/odo/blob/master/pkg/occlient/occlient.go#L72

Unfortunately this is hard-coded. I've increased the seconds to 120 and it seems to decrease the bug occurring. However, we should be able to set this manually within `odo utils config`

Ping @kadel @anmolbabu @surajnarwade

**Comment by kadel** [comment]
OK, it makes sense to have this configurable, but lets call this option as something that is more user-friendly.  How about `pushTimout`?

**Comment by kadel** [comment]
After this is implemented, we can remove what has been done in #943

**Comment by cdrage** [comment]
Just implementing this now, but would this be better in `odo preference` rather than `odo config` since preference is a global file (in the users directory?).

If implemented via `odo config` (which the configuration file follows the folder where the components are), we would have to modify the `occlient.go` New() function: https://github.com/openshift/odo/blob/master/pkg/occlient/occlient.go#L225 so that we pass in cobra cmd commands, in order to get the context, changing the functionality of `occlient.go` into a function that interpolates with the CLI.

TLDR; Go with preference, easy implementation. Go with config, it will change the occlient functionality.

Thoughts @girishramnani @kadel ?

**Comment by kadel** [comment]
It should definitely be a preference. I think that the original description mentions `odo  config` because it was created before we had `odo preference` (before new workflow). At that time. `odo config` served the same purpose as the current `odo preference` - global settings applied to every `odo` command invoked by user.
